lib/normalizer.c

```diff
@@ -845,11 +845,79 @@ grn_nfkc_normalize_expand(grn_ctx *ctx,
                                     "");
 }
 
+/*
+ * This function assumes that the input utf8_char is a valid UTF-8 character.
+ * It is the caller's responsibility to ensure that utf8_char is valid UTF-8
+ * before calling this function. This is a precondition in terms of contract
+ * programming.
+ */
 grn_inline static const unsigned char *
 grn_nfkc_normalize_unify_alphabet_diacritical_mark(
   const unsigned char *utf8_char, unsigned char *unified)
 {
-  // TODO: Implement function to return `unified` instead of `utf8_char`.
+  /*
+   * Latin-1 Supplement
+   * U+00E0 LATIN SMALL LETTER A WITH GRAVE ..
+   * U+00E5 LATIN SMALL LETTER A WITH RING ABOVE
+   */
+  if ((utf8_char[0] == 0xC3 && 0xA0 <= utf8_char[1] && utf8_char[1] <= 0xA5) ||
+      /*
+       * Latin Extended-A
+       * U+0101 LATIN SMALL LETTER A WITH MACRON
+       * U+0103 LATIN SMALL LETTER A WITH BREVE
+       * U+0105 LATIN SMALL LETTER A WITH OGONEK
+       * Uppercase counterparts(U+0102 and U+0104) are omitted as NFKC
+       * normalization converts them to their lowercase equivalents.
```

Comment:
```suggestion
       * Uppercase counterparts (U+0102 and U+0104) are covered by the following
       * condition but they are never appeared here. Because NFKC normalization
       * normalization converts them to their lowercase equivalents.
```

Reply:
fix: [38cf53f](https://github.com/groonga/groonga/pull/1859/commits/38cf53f30c6e7040205ff796458426a1141f2aa0) ありがとうございます。なぜマッチしている範囲に含まれているのに書かれていないの関係の部分がぬけていましたねmm